### ISSUE TITLE
Fix typo in SMIRNOFF spec and in AlkEthOH FFXML files

### ISSUE THREAD

**Issue by jchodera** [open-forcefield-group/openforcefield#17]
In the [Nonbonded Parameters](https://github.com/open-forcefield-group/openforcefield/blob/master/The-SMIRNOFF-force-field-format.md#nonbonded-parameters-eg-lennard-jones) section of the spec, we have the example
```XML
   <Atom smirks="[$([#1]-[#6]):1]" rmin_half="1.4870" epsilon="0.0157"/>
```
This SMIRKS string doesn't seem sensible to me. Which atom does the `:1` tag in this case---the hydrogen or carbon?

**Comment by cbayly13** [comment]
I don't know what that SMIRKS means and it doesn't appear in my smirff99Frosst.ffxml from last September. What I do see in my file is a single occurrence of the rmin_half value of 1.4870 in `<Atom smirks="[#1:1]-[#6X4]" epsilon="0.0157" id="n3" rmin_half="1.4870"/>`, so I think I know what the spec is supposed to have.

**Comment by cbayly13** [comment]
Or perhaps the spec is just proposing a generic for H-C which could be:
`<Atom smirks="[#1:1]-[#6]" epsilon="0.0157" id="n3" rmin_half="1.4870"/>`
or if we want to use the "$":
`<Atom smirks="[#1$(*-[#6]):1] ]" epsilon="0.0157" id="n3" rmin_half="1.4870"/>`

**Comment by davidlmobley** [comment]
This was from the AlkEthOH FFXML, @cbayly13 . I think it should be the generic though:
`<Atom smirks="[#1:1]-[#6]" epsilon="0.0157" id="n3" rmin_half="1.4870"/>`

I'm guessing maybe someone mistyped this at some point for AlkEthOH and then it accidentally got copied into the spec. It actually had no functional relevance in AlkEthOH since the parameters were the same as for the generic hydrogen, IIRC.

**Comment by davidlmobley** [comment]
It looks to me from the GitHub history that this was introduced as a typo in the original FFXML file when @jchodera hand-converted it from Chris Bayly's text mockups, specifically: 

https://github.com/open-forcefield-group/smarty/blob/3d02bc8ab0911b3353b38ba2c08418eddc59420e/smarty/data/forcefield/Frosst_AlkEtOH.ffxml#L38

From there it got eventually adjusted to replace the "C" with the element number for consistency with the rest of the FFXML but, because it had no functional importance (since it uses the same parameters as the generic hydrogen) we never caught that the SMIRKS was "odd". The SMIRKS is VALID (in the sense that it matches things) so it doesn't raise any issues on that front. 

This is a problem both with the documentation and with the several AlkEthOH FFXML files. I'll change the name of this issue and we can fix to `<Atom smirks="[#1:1]-[#6]" epsilon="0.0157" id="n3" rmin_half="1.4870"/>`.

**Comment by cbayly13** [comment]
I have looked back and I see `[$([#1]-C)]` in my earliest mockups for AlkEthOH so the blame falls on me for where it comes from. 
It could be that `[$([#1]-[#6]):1]` actually makes sense; to remind myself of that syntax:
in PICTO I can't see hydrogens so I put in tetrahydropyran `C1CCOCC1` and when I entered the query `[$([#8]-[#6])]` it matched the oxygen, so maybe the SMIRKS syntax is OK. 
I must say I prefer `[#1:1]-[#6]` for readability.

**Comment by davidlmobley** [comment]
Ah. That would fit with my testing too (it seemed to me the syntax was OK but I couldn't make any sense of it). I'll fix to correct the readability.

**Comment by bannanc** [comment]
In recursive SMIRKS (the ones with $(stuff) ) the first atom inside the brackets refers to the current atom.  So `[$([#1]-[#6]):1]` is a hydrogen bound to a carbon, and the first atom in the embedded string will get the index. It is equivalent to `"[#1$(*-[#6]):1] ]"` and `[#1:1]-[#6]` for the purposes of identifying the hydrogen. Chemical Environments can parse these and will make a hydrogen as atom1 with an alpha atom that is carbon.

**Comment by davidlmobley** [comment]
Great, @bannanc . But you have no objections to us swapping this to the easier-to-read form?

**Comment by bannanc** [comment]
Oh no, easier to read is definitely better.

**Comment by jchodera** [comment]
Thanks everyone!  

It might be good to take a careful read-through the spec to make sure there are no other accidental typos.

**Comment by bannanc** [comment]
I replaces the SMIRKS that started this discussion in PR #40 
I also skimmed through `The-SMIRNOFF-force-field-format.md` the rest of the SMIRKS strings there seem more readable/normal as in they have the indexed atoms labeled directly, no recursive SMIRKS. I haven't read all of our documentation to check for this are there other places we should be concerned about?

**Comment by davidlmobley** [comment]
@bannanc - check the tests directory; there might be some example SMIRKS with this format in tests. You might also just search (search bar at the top) or even Google for the offending SMIRKS to see if it's hanging around somewhere.

**Comment by bannanc** [comment]
For everyone's future reference, if you want to search for SMIRKS you have to grep for the "fixed" string so the best command option is `git grep -Fn 'SMIRKS STRING'`

**Comment by bannanc** [comment]
The particular SMIRKS doesn't show up other than in some of the AlkEthOH ffxml files. I did some searching on README files and didn't find any SMIRKS starting with `'[$('` so I think this issue can be closed when PR #40 is accepted.
